.github/workflows/update-caches.yml

```diff
@@ -0,0 +1,136 @@
+name: Update Actions Caches
+
+on:
+  # Manually
+  workflow_dispatch:
+  # On PR merge
+  push:
+    branches:
+      - main
+  # After nightly release
+  schedule:
+    - cron: "0 1 * * *"
+
+env:
+  CARGO_INCREMENTAL: 0
```

Comment:
```suggestion
```

(Nit) `dtolnay/rust-toolchain` already [does this for us](https://github.com/dtolnay/rust-toolchain/blob/b3b07ba8b418998c39fb20f53e8b695cdcc8de1b/action.yml#L104-L109) :)